Content.Shared/GameObjects/Components/Weapons/Ranged/SharedRangedBarrelComponent.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.GameObjects.Components.Weapons.Ranged
+{
+    public abstract class SharedRangedBarrelComponent : Component
+    {
+        public abstract FireRateSelector FireRateSelector { get; }
+        public abstract FireRateSelector AllRateSelectors { get; }
+        public abstract float FireRate { get; }
+        public abstract int ShotsLeft { get; }
+        public abstract int Capacity { get; }
+    }
+
+    [Flags]
+    [FlagsFor(typeof(FireRateSelector))]
```

Comment:
Unless `FireRateSelector` somehow appears in the engine code, this shouldn't be necessary. It's an attribute designed for `int`s in the engine that have some special flag names defined in the content, like collision layers.

Reply:
Isn't it required for flags support in the yaml or is there another way to do it? I tried removing it but it started throwing.

Reply:
It is not required for that, no. It's only required if you have something which is an `int` in the code, but you want to treat it as an `enum` in the YAML. And typically that only happens when you're passing something to the engine.